### DISCUSSION TITLE
Enum values that differ in case get conflated

### DISCUSSION THREAD

**Discussion by krzysztofwos** [fastapi/typer#570] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from enum import Enum

import typer


class Interval(Enum):
    ONE_MINUTE = "1m"
    ONE_MONTH = "1M"


def main(interval: Interval = typer.Argument("1m", case_sensitive=True)):
    print(interval)


if __name__ == "__main__":
    typer.run(main)
```

### Description

The example illustrates the problem: enum values that differ in case get conflated. In the example above, running the script with `1m` and `1M` results in the `main` function receiving `Interval.ONE_MONTH`. The issue starts in the function below, lifted from `main.py`:

```python
def generate_enum_convertor(enum: Type[Enum]) -> Callable[[Any], Any]:
    lower_val_map = {str(val.value).lower(): val for val in enum}

    def convertor(value: Any) -> Any:
        if value is not None:
            low = str(value).lower()
            if low in lower_val_map:
                key = lower_val_map[low]
                return enum(key)

    return convertor
```

### Operating System

Linux

### Typer Version

0.7.0

### Python Version

3.10.6

**Comment by avaldebe** [comment, marked as answer]
This is a bug 🐞, IMO. I submitted a PR to fix it #571. Hope it gets merged...

**Reply by tiangolo** [reply]
Indeed, this was a bug, thanks @avaldebe for the fix! It will be available in Typer 0.9.1 in the next few hours.
